quiche/src/lib.rs

```diff
@@ -7418,6 +7423,7 @@ impl<F: BufFactory> Connection<F> {
                         lost_bytes,
                         acked_bytes,
                         spurious_losses,
+                        startup_exit_reason,
```

Comment:
I would consider just keeping this state inside recovery rather than returning it from on_ack_received.

Also, I think we could exit slow start during an RTO instead of ack processing instead of ack processing.

Reply:
Yea good point about other times we can exit startup and your suggestion makes sense.

I dont love keeping pure stat state inside recovery because it adds bloat (eventually I would like to make this a user opt-in functionality).

Reply:
> I would consider just keeping this state inside recovery rather than returning it from on_ack_received.

I think the return pattern is more type safe because it makes consume side-effects obvious and makes sure we consume them.

Return value:
"on_ack_received returns on_startup_exit reason. It might have recorded a value so make sure to consume it."

vs

Explicit API:
"The on_ack_received might have modified the on_ack_received value. You have to explicitly know that on_ack_received potentially records a new value and call a separate API `recovery.get_startup_exit_reason()`."

Reply:
Consider the following:
* Slow start is done on each path, so the right place to expose this is from `PathStats` as Alessandro points out.
* You are forced to remember to look at this output argument in both on_ack_received and on_loss_detection_timeout, not just 1 place
* The startup_exit_reason remains constant past the point at which it changes
* Significant prior art of getting info from the recovery module in `PathStats` accessor https://github.com/cloudflare/quiche/blob/master/quiche/src/path.rs#L504